test/bottomFriction/test_ekman_bottom.py

```diff
@@ -0,0 +1,116 @@
+"""
+Bottom Ekman layer test
+=======================
+
+Steady state flow in a channel subject to bottom friction and rotation.
+Vertical viscosity is assumed to be constant to allow simple analytical
+solution.
+"""
+from thetis import *
+import numpy
+import pytest
+
+
+def run_test(layers=25, tolerance=0.05, verify=True, **model_options):
+    depth = 20.0
+    surf_slope = -5.0e-6  # d elev/dx
+
+    # set mesh resolution
+    dx = 2500.0
+    nx = 3
+    lx = nx*dx
+    ny = 3
+    ly = ny*dx
+    mesh2d = PeriodicRectangleMesh(nx, ny, lx, ly, direction='both',
+                                   reorder=True)
+
+    dt = 90.0
+    t_end = 4 * 3600.0  # sufficient to reach ~steady state
+    t_export = 450.0
+    u_mag = 1.0
+
+    f_coriolis = 1e-4
+    nu_v = 5e-4
+
+    if os.getenv('THETIS_REGRESSION_TEST') is not None:
+        t_end = 5*t_export
+
+    # bathymetry
+    p1_2d = get_functionspace(mesh2d, 'CG', 1)
+    bathymetry2d = Function(p1_2d, name='Bathymetry')
+    bathymetry2d.assign(depth)
+
+    # create solver
+    solver_obj = solver.FlowSolver(mesh2d, bathymetry2d, layers)
+    options = solver_obj.options
+    options.element_family = 'dg-dg'
+    options.timestepper_type = 'SSPRK22'
+    options.solve_salinity = False
+    options.solve_temperature = False
+    options.use_implicit_vertical_diffusion = True
+    options.use_bottom_friction = True
+    options.bottom_roughness = Constant(1e-3)
+    options.use_turbulence = False
+    options.coriolis_frequency = Constant(f_coriolis)
+    options.vertical_viscosity = Constant(nu_v)
+    options.vertical_diffusivity = Constant(nu_v)
+    options.simulation_export_time = t_export
+    options.timestepper_options.use_automatic_timestep = False
+    options.timestep = dt
+    options.simulation_end_time = t_end
+    options.horizontal_velocity_scale = Constant(u_mag)
+    options.no_exports = True
+    options.update(model_options)
+
+    solver_obj.create_function_spaces()
+
+    # drive flow with momentum source term equivalent to constant surface slope
+    pressure_grad = -physical_constants['g_grav'] * surf_slope
+    options.momentum_source_2d = Constant((pressure_grad, 0))
+
+    solver_obj.create_equations()
+
+    v_init_2d = -0.49
+    solver_obj.assign_initial_conditions(uv_2d=Constant((0, v_init_2d)))
+
+    x, y, z = SpatialCoordinate(solver_obj.mesh)
+
+    solver_obj.iterate()
+
+    if verify and os.getenv('THETIS_REGRESSION_TEST') is None:
```

Comment:
This is a little odd: what is this test testing that is not tested by the example (through test_examples.py)? and why do we need THETIS_REGRESSION_TEST logic here?

Reply:
Oh, thanks for pointing that out. I must have copied that over from the examples.